### ISSUE TITLE
Aztec crashes on Emoji

### ISSUE THREAD

**Issue by koke** [wordpress-mobile/WordPress-iOS#6732]
### Steps to reproduce the behavior

- Open a new post
- Type an emoji character 🤔, an "invisible" character is added to the text
- Type another emoji 🤧
- Crash

`TextStorage` crashes here:

```swift
private func hasNewLine(atIndex index: Int) -> Bool {
    let swiftStringIndex = textStore.string.index(textStore.string.startIndex, offsetBy: index)

    return string.characters[swiftStringIndex] == Character(.newline)
}
```

```
(lldb) po swiftStringIndex
▿ Index
  ▿ _base : Index
    - _position : 2
  - _countUTF16 : 0


(lldb) po textStore.string
"🤔"
```

We seem to be using an index that counts UTF16 code units for a `CharacterView` which contains _ extended grapheme clusters_.

🤔 in UTF-16 takes two characters (0xD83E 0xDD14)

**Comment by jleandroperez** [comment]
Addressed in the Aztec project (and WPiOS got the latest via #6863).

The editor no longer crashes on emoji insertion. Thanks @koke!!!

**Comment by jeremeylduvall** [comment]
@jleandroperez I think this is back again 🙁 The latest version of the app (7.8.0.20170605) is crashing whenever I try to enter an emoji into the text field (not the title field).

![smiley](https://user-images.githubusercontent.com/7240478/27189340-5992c7c2-51ae-11e7-81a0-2a19d8690d19.gif)

**Comment by jleandroperez** [comment]
@jeremeylduvall thanks for the report sir!! Addressed via #7382 ("Major Aztec Overhaul")

Should be fixed in the next Internal Beta!
